compiler/qsc_parse/src/item.rs

```diff
@@ -44,6 +44,12 @@ pub(super) fn parse(s: &mut Scanner) -> Result<Box<Item>> {
     } else if let Some(callable) = opt(s, parse_callable_decl)? {
         Box::new(ItemKind::Callable(callable))
     } else {
+        if !doc.is_empty() {
+            return Err(Error(ErrorKind::FloatingDocComment(Span {
```

Comment:
The `Span` used here seems irrelevant to the error message - it might be more helpful if we used the doc comment's span instead of the next token's span.

Reply:
I'll have to see if there is a way to directly get the span of the doc comment. What I have here _is_ getting the span (at least for the examples I've ran it on), just indirectly.

Reply:
Update: I was able to get the span directly thanks again